data_model/src/trigger/set.rs

```diff
@@ -120,14 +120,61 @@ impl Set {
         self.ids.iter().map(|entry| entry.key().clone()).collect()
     }
 
+    /// Apply `f` to triggers that belong to the given [`DomainId`]
+    ///
+    /// Returns an empty list if [`Set`] doesn't contain any trigger belonging to [`DomainId`].
+    pub fn inspect_by_domain_id<F, R>(&self, domain_id: &DomainId, f: F) -> Vec<R>
+    where
+        F: Fn(&Id, Action<FilterBox>) -> R,
```

Comment:
```suggestion
        F: Fn(&Id, &dyn ActionTrait) -> R,
```

This is how it's done in `inspect_by_id()`. Why not to use the same technique here to prevent extra cloning

Reply:
I thought about it but can't compile it due to lifetimes. That's why I added boxing. Actually, it will box and clone anyway eventually when calling this function.